pkg/image/importer/importer.go

```diff
@@ -299,9 +319,35 @@ func formatRepositoryError(repository *importRepository, refName string, refID s
 	return
 }
 
+func calculateImageSize(ctx gocontext.Context, repo distribution.Repository, image *api.Image, cache *imageStreamImporterCache) error {
+	bs := repo.Blobs(ctx)
+
+	size := int64(0)
+	for i := range image.DockerImageLayers {
+		layer := &image.DockerImageLayers[i]
+
+		if layerSize, ok := cache.LayerSizeCache[layer.Name]; ok {
+			size += layerSize
+			continue
+		}
+
+		desc, err := bs.Stat(ctx, digest.Digest(layer.Name))
```

Comment:
@soltysh Actually, I'm not sure this is right. The `calculateImageSize` is called after `ImageWithMetadata`. If after calling the `ImageWithMetadata`, the metadata size is zero, it means we can't trust the layers size and we have to use `Stat`.